File: apps/mobile/targets/positions-widget/Sources/Utilities/ChainCatalog.swift
import Foundation

enum ChainCatalog {
  static func name(for chainID: Int) -> String {
    switch chainID {
    case 1: return "Ethereum"
    case 8453: return "Base"
    case 42161: return "Arbitrum"
    case 10: return "Optimism"
    case 137: return "Polygon"
    case 56: return "BNB Chain"
    case 43114: return "Avalanche"
    default: return "Chain \(chainID)"
    }
  }

  static func shortName(for chainID: Int) -> String {
    switch chainID {
    case 1: return "ETH"
    case 8453: return "Base"
    case 42161: return "ARB"
    case 10: return "OP"
    case 137: return "MATIC"
    case 56: return "BNB"
    case 43114: return "AVAX"
    default: return "\(chainID)"
    }
  }
}

File: apps/mobile/modules/widget-bridge/android/src/main/java/expo/modules/widgetbridge/widget/util/ChainCatalog.kt
package expo.modules.widgetbridge.widget.util

object ChainCatalog {
  fun name(chainId: Int): String = when (chainId) {
    1 -> "Ethereum"
    8453 -> "Base"
    42161 -> "Arbitrum"
    10 -> "Optimism"
    137 -> "Polygon"
    56 -> "BNB Chain"
    43114 -> "Avalanche"
    else -> "Chain $chainId"
  }

  fun shortName(chainId: Int): String = when (chainId) {
    1 -> "ETH"
    8453 -> "Base"
    42161 -> "ARB"
    10 -> "OP"
    137 -> "POL"
    56 -> "BNB"
    43114 -> "AVAX"
    else -> chainId.toString()
  }
}
